test/framework/fixtures.tmpl

```diff
@@ -0,0 +1,209 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: {{.Namespace}}
+resources:
+- ../../../../test/testdata/fixtures/{{ .Fixture }}
+
+patches:
+{{ if .K8ssandraCluster }}
+{{ with .K8ssandraCluster }}
+  - target:
+      kind: K8ssandraCluster
+    patch: |-
+
+      - op: replace
+        path: /spec/auth
+        value: {{ if .Auth }}{{ .Auth }}{{ else }}true{{ end }}
+
+  {{ if and $.Storage .Cassandra.StorageConfig }}
```

Comment:
Is it possible to avoid this additional templating? Could we instead just have kustomizations for each cloud provider that applies whatever changes necessary? Kustomize can be a PITA, but from a usability perspective, I think it would be a win in that users could simply run `kustomize build` to see an example manifest for GKE.

Reply:
I think we can indeed have a few examples of kustomizations for each cloud provider. But to use those examples in our e2e tests, those kustomizations would have to contain the actual external IPs to use, and the actual zone names (racks). This would be undesirable IMO because 1) the external IPs are kind of sensitive and I'd avoid to expose them, and 2) this info could change over time if we re-create the test cluster with different IPs or different zones. One possible solution would be to investigate if kustomize can accept environment variables for these bits, which would allow us to avoid hard-coding them in the kustomize file.

Reply:
>the external IPs are kind of sensitive and I'd avoid to expose them

Agreed. They could be provide to GHA via secrets.

>this info could change over time if we re-create the test cluster with different IPs or different zones

I'm not too concerned about the zone labels. We don't have to use the zone labels. We can use whatever labels we want. We can configured labels on worker nodes with whatever automation we use to configure the k8s clusters.

Let's find some time to sync up on this. I would at least like to have some idea how things would look with a more declarative approach vs the templating-heavy approach.